projects/aca-content/src/lib/components/search/search-input/search-input.component.ts

```diff
@@ -121,14 +120,14 @@ export class SearchInputComponent implements OnInit, OnDestroy {
   ngOnInit() {
     this.showInputValue();
 
-    this.router.events
-      .pipe(takeUntil(this.onDestroy$))
-      .pipe(filter((e) => e instanceof RouterEvent))
-      .subscribe((event) => {
-        if (event instanceof NavigationEnd) {
-          this.showInputValue();
-        }
-      });
+    this.route.queryParams.pipe(takeUntil(this.onDestroy$)).subscribe((params: Params) => {
+      const encodedQuery = params['q'] ? params['q'] : null;
```

Comment:
I think it’s worth to put these two lines of code to inside of ‘if’ as we don’t need to execute that code if we won’t enter if body

Reply:
@dominikiwanekhyland I think you misunderstood me here, I meant following: 
<img width="851" alt="image" src="https://github.com/user-attachments/assets/316842ae-8001-4b21-a9f8-60990d5f3076">

You can see that in second if we assign searchedWord which is calculated outside if, so I meant that it has sense probably to put whole code which is inside subscribe to inside that if as we don't need to calculate searchedWord if there is no searchInputControl I guess?